arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/callbacks.kt

```diff
@@ -0,0 +1,123 @@
+package arrow.fx.coroutines.stream
+
+import arrow.core.Either
+import arrow.fx.coroutines.CancelToken
+import arrow.fx.coroutines.ExitCase
+import arrow.fx.coroutines.ForkConnected
+import arrow.fx.coroutines.UnsafePromise
+import arrow.fx.coroutines.guaranteeCase
+import arrow.fx.coroutines.stream.concurrent.Queue
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.startCoroutine
+import kotlin.experimental.ExperimentalTypeInference
+
+private object END
+
+interface EmitterSyntax<A> {
+  fun onCancel(cancelF: suspend () -> Unit): Unit
+  fun emit(a: A): Unit
+  fun emit(chunk: Chunk<A>): Unit
+  fun emit(iterable: Iterable<A>): Unit
```

Comment:
Arrow Fx Coroutine Stream works differnt from KotlinX Coroutine Flow, both are `suspend` pull based, but Arrow Fx's Stream works `Chunk` based. So `emit(a: A)` is actually the special case here, since that emits a singleton `Chunk`.

I don't think we should be concerned about matching other libraries API :D